src/Sylius/Bundle/ApiBundle/DataProvider/CartPaymentMethodsSubresourceDataProvider.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\DataProvider;
+
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use ApiPlatform\Core\DataProvider\SubresourceDataProviderInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Core\Repository\PaymentRepositoryInterface;
+use Sylius\Component\Payment\Resolver\PaymentMethodsResolverInterface;
+use Webmozart\Assert\Assert;
+
+final class CartPaymentMethodsSubresourceDataProvider implements RestrictedDataProviderInterface, SubresourceDataProviderInterface
+{
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var PaymentRepositoryInterface */
+    private $paymentRepository;
+
+    /** @var PaymentMethodsResolverInterface */
+    private $paymentMethodsResolver;
+
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        PaymentRepositoryInterface $paymentRepository,
+        PaymentMethodsResolverInterface $paymentMethodsResolver
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->paymentRepository = $paymentRepository;
+        $this->paymentMethodsResolver = $paymentMethodsResolver;
+    }
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        $subresourceIdentifiers = $context['subresource_identifiers'] ?? null;
+
+        return
+            is_a($resourceClass, PaymentMethodInterface::class, true) &&
+            isset($subresourceIdentifiers['id'], $subresourceIdentifiers['payments'])
+        ;
+    }
+
+    public function getSubresource(string $resourceClass, array $identifiers, array $context, string $operationName = null)
+    {
+        $subresourceIdentifiers = $context['subresource_identifiers'];
+
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findCartByTokenValue($subresourceIdentifiers['id']);
+        Assert::notNull($order);
+
+        /** @var PaymentInterface|null $payment */
+        $payment = $this->paymentRepository->find($subresourceIdentifiers['payments']);
+        Assert::notNull($payment);
+
+        Assert::true($order->hasPayment($payment), 'Payment doesn\'t match for order');
+
+        if (!$order->hasPayments()) {
+            return [];
+        }
```

Comment:
If we will pass through the Assert, `if` statement is always false.
```suggestion
        Assert::true($order->hasPayment($payment), 'Payment doesn\'t match for order');
```